File: helpers/bandchain2.js/__tests__/data.test.ts
import { Coin } from '../src/data'

describe('Coin', () => {
  it('success', () => {
    let coin = new Coin(1000, 'uband')

    expect(coin.validate()).toBeTruthy()

    const exepected = {
      amount: '1000',
      denom: 'uband',
    }

    expect(coin.asJson()).toEqual(exepected)
  })

  it('error integer', () => {
    let coin = new Coin(1000.2, 'uband')
    expect(() => {
      coin.validate()
    }).toThrowError('amount is not an integer')
  })

  it('error amount less than 0', () => {
    let coin = new Coin(-50, 'uband')
    expect(() => {
      coin.validate()
    }).toThrowError('Expect amount more than 0')
  })

  it('error expect denom', () => {
    let coin = new Coin(1000, '')
    expect(() => {
      coin.validate()
    }).toThrowError('Expect denom')
  })
})

File: helpers/pyband/tests/data_test.py
import pytest
from pyband.data import Coin
from pyband.exceptions import InsufficientCoinError


def test_coin_success():
    coin = Coin(amount=5000, denom="uband")

    assert coin.as_json() == {"amount": "5000", "denom": "uband"}

    assert coin.validate() == True


def test_coin_amount_fail():
    coin = Coin(amount=-5000, denom="uband")

    with pytest.raises(InsufficientCoinError, match="Expect amount more than 0"):
        coin.validate()


def test_coin_denom_fail():
    coin = Coin(amount=5000, denom="")

    with pytest.raises(ValueError, match="Expect denom"):
        coin.validate()
